src/parsing_routines_matrixnetworks.jl

```diff
@@ -122,3 +122,103 @@ function loadrxnetwork(ft::MatrixNetwork,
+         1, if the i'th complex is the product of the j'th reaction,
+         0, otherwise
+"""
+#  for dense matrices
+function LoadReacCompNetwork(rateexprs::AbstractVector,
+                        compstoichmat::AbstractMatrix,
+                        incidencemat::AbstractMatrix;
+                        species::AbstractVector = Any[],
+                        params::AbstractVector = Any[])
+
+    numspecs, numcomp = size(compstoichmat)
+    @assert all(>=(0),compstoichmat)
+    @assert numcomp == size(incidencemat, 1)
+    @assert all(∈([-1,0,1]),incidencemat)
+    numrxs = size(incidencemat, 2)
+
+    ModelingToolkit.@parameters t
+    isempty(species) && (species = [funcsym(:S,t,i) for i = 1:numspecs])
+
+    rn = Vector{Reaction}(undef,numrxs)
+    sub_indices = argmin(incidencemat, dims=1)     # cartesian indices of substrate complexes
+    prod_indices = argmax(incidencemat, dims=1)    # cartesian indicies of products complexes
+
+    for i ∈ 1:numrxs
+        subStoichInd = getindex.(findall(!iszero, compstoichmat[:,sub_indices[i][1]]))
+        prodStoichInd = getindex.(findall(!iszero, compstoichmat[:,prod_indices[i][1]]))
+
+        if subStoichInd == Int64[] &&  prodStoichInd != Int64[]
+            rn[i] = Reaction(rateexprs[i], nothing, species[prodStoichInd],
+                        nothing, compstoichmat[prodStoichInd,prod_indices[i][1]])
+
+        elseif subStoichInd != Int64[] &&  prodStoichInd == Int64[]
+            rn[i] = Reaction(rateexprs[i], species[subStoichInd], nothing,
+                        compstoichmat[subStoichInd,sub_indices[i][1]], nothing)
+        else
+            rn[i] = Reaction(rateexprs[i], species[subStoichInd],species[prodStoichInd],
+                        compstoichmat[subStoichInd,sub_indices[i][1]],
+                        compstoichmat[prodStoichInd,prod_indices[i][1]])
+        end
+    end
+
+    @named rs = ReactionSystem(rn, t, species, params)
+end
+
+
+#  for sparse matrices
+function LoadReacCompNetwork(rateexprs::AbstractVector,
+                        compstoichmat::SparseMatrixCSC,
+                        incidencemat::SparseMatrixCSC;
+                        species::AbstractVector = Any[],
+                        params::AbstractVector = Any[])
+
+    numspecs, numcomp = size(compstoichmat)
+    @assert all(>=(0),compstoichmat)
+    @assert numcomp == size(incidencemat, 1)
+    @assert all(∈([-1,0,1]),incidencemat)
+    numrxs = size(incidencemat, 2)
+
+    ModelingToolkit.@parameters t
+    isempty(species) && (species = [funcsym(:S,t,i) for i = 1:numspecs])
+
+    rn = Vector{Reaction}(undef,numrxs)
+
+    sub_indices = argmin(incidencemat, dims=1)     # cartesian indices of substrate complexes
+    prod_indices = argmax(incidencemat, dims=1)    # cartesian indicies of products complexes
+
+    for i ∈ 1:numrxs
+        subStoichInd = getindex.(findall(!iszero, compstoichmat[:,sub_indices[i][1]]))
```

Comment:
I think you should be using the sparsity of `compstoichmat` here and searching it in that way. It doesn't make sense to allocate a dense vector for a column from a sparse matrix.

Reply:
resolved..used `rowvals` and `colptr` fields from `sparsematrix`